sequencer/src/header.rs

```diff
@@ -218,18 +218,19 @@ impl Header {
             builder_signature: None,
         };
 
-        // Sign Header with builder wallet from state
-        let signing_key: &SigningKey = builder_address.signer();
-        let header_signature: ecdsa::Signature = signing_key.sign(header.commit().as_ref());
-        // In order to use `types::Signature` for verification we do a
-        // `TryFrom` conversion on the signature. But first we need to
-        // append Etherium's `Electurm` value (can be 27 or 28). There
-        // must be a better way to do this.
-        let header_signature = [header_signature.to_vec(), vec![27]].concat();
+        // Sign our header using its `Committment` as a prehash.
+        let builder_signature = builder_address
+            .sign_hash(types::H256(header.commit().into()))
+            .unwrap();
```

Comment:
Not sure about this unwrap. I can't think of why a signature would fail, but it isn't documented and maybe there is some weird case where it does. Anyways, since `Header::new` currently doesn't return a result, there's not much we can do here *but* panic. Although maybe it would be cleaner to make `from_info` return a result and then just have a top level `unwrap` in `Header::new` with a comment explaining we should fix this when HotShot is updated to [handle failures in block building](https://espresso.zulipchat.com/#narrow/stream/311646-Consensus/topic/New.20failure.20modes)